server/src/gameMachine/index.ts

```diff
@@ -0,0 +1,303 @@
+import {
+  removeKey,
+  type BaseUnifiedGame,
+  toEntries,
+  type Role,
+  shuffleList,
+  type Alignment,
+  type PlayerStatus,
+} from "@hidden-identity/shared";
+import {
+  configureStore,
+  type EnhancedStore,
+  type StoreEnhancer,
+  type ThunkDispatch,
+  type UnknownAction,
+} from "@reduxjs/toolkit";
+import { UNASSIGNED } from "../database/gameDB/base.ts";
+
+// Needed so things can read the type of gameStore
+export type { EnhancedStore, StoreEnhancer, ThunkDispatch, UnknownAction };
+
+export const gameStore = configureStore<BaseUnifiedGame, AnyAction>({
+  reducer: {
+    playersToRoles: (state = {}, action) => {
+      switch (action.type) {
+        case "AddPlayer":
+          return {
+            ...state,
+            [action.player]: UNASSIGNED,
+          };
+        case "KickPlayer":
+          return removeKey(state, action.player);
+        case "ChangePlayerRole":
+        case "DrawRole":
+          return {
+            ...state,
+            [action.player]: action.role,
+          };
+        default:
+          return state;
+      }
+    },
+
+    partialPlayerOrdering: (state = {}, action) => {
+      switch (action.type) {
+        case "AddPlayer":
+          return {
+            ...state,
+            [action.player]: { rightNeighbor: null },
+          };
+        case "SetNeighbor": {
+          return {
+            ...state,
+            [action.player]: { rightNeighbor: action.newRightNeighbor },
+          };
+        }
+
+        // When kicking, extract from the circle
+        case "ExtractPlayerFromCircle":
+        case "KickPlayer": {
+          const { [action.player]: kickedPlayerNeighbor, ...nextState } = state;
+          const oprhanedNeighbor = toEntries(nextState).find(
+            ([_, value]) => value?.rightNeighbor === action.player,
+          );
+
+          if (oprhanedNeighbor) {
+            return {
+              ...nextState,
+              [oprhanedNeighbor[0]]: {
```

Comment:
Very nitpicky, but I'd prefer orphaned neighbor was desctructed from the get.
Also typo "oprhaned"

Reply:
Can't it might be null